src/model.rs

```diff
@@ -927,6 +953,40 @@ macro_rules! impl_ProblemOrSolving {
                 cons
             }
 
+            /// Adds a new indicator constraint to the model with the given variables, coefficients, right-hand side, and name.
+            ///
+            /// # Arguments
+            ///
+            /// * `bin_var` - The binary variable in the constraint.
+            /// * `vars` - The variables of the constraints.
+            /// * `coefs` - The coefficients of the variables in the constraint.
+            /// * `rhs` - The right-hand side of the constraint.
+            /// * `name` - The name of the constraint.
+            ///
+            /// # Returns
+            ///
+            /// A reference-counted pointer to the new constraint.
+            ///
+            /// # Panics
+            ///
+            /// This method panics if the constraint cannot be created in the current state.
+            fn add_cons_indicator(
+                &mut self,
+                bin_var: Rc<Variable>,
+                vars: Vec<Rc<Variable>>,
+                coefs: &mut [f64],
+                rhs: f64,
+                name: &str,
+            ) -> Rc<Constraint> {
+                assert_eq!(vars.len(), coefs.len());
```

Comment:
Would be good to have an assert here that the `bin_var` is indeed binary.

Reply:
Added the assert.